crates/js-component-bindgen/src/intrinsics.rs

```diff
@@ -114,6 +119,117 @@ pub fn render_intrinsics(
 
     for i in intrinsics.iter() {
         match i {
+            Intrinsic::AsyncifyAsyncInstantiate => output.push_str("
+                const asyncifyModules = [];
+                let asyncifyPromise;
+                let asyncifyResolved;
+                async function asyncifyInstantiate(module, imports) {
+                  const instance = await instantiateCore(module, imports);
+                  const memory = instance.exports.memory || (imports && imports.env && imports.env.memory);
```

Comment:
I think we should be a bit smarter about this memory detection, and instead make the `memoryExportName` an argument into the `asyncifyInstantiate` function - we should be able to have the context at the Rust side to know this information?

Also flipping between the imports or the exports for the memory seems a bit too loose - can you explain the cases further? Could we make it a boolean argument from the Rust side whether the import or export memory should be used?